libsolidity/inlineasm/AsmData.h

```diff
@@ -49,9 +49,11 @@ struct Assignment { SourceLocation location; Identifier variableName; };
 struct FunctionalAssignment;
 struct VariableDeclaration;
 struct FunctionDefinition;
+struct Switch;
 struct FunctionCall;
 struct Block;
-using Statement = boost::variant<Instruction, Literal, Label, Assignment, Identifier, FunctionalAssignment, FunctionCall, FunctionalInstruction, VariableDeclaration, FunctionDefinition, Block>;
+using Statement = boost::variant<Instruction, Literal, Label, Assignment, Identifier, FunctionalAssignment, FunctionCall, FunctionalInstruction, VariableDeclaration, FunctionDefinition, Switch, Block>;
+using Expression = boost::variant<FunctionCall, Identifier, Literal>;
```

Comment:
I had no success in keeping it as `Statement` as it had a recursive issue then. I guess it had to be wrapper in `shared_ptr` for that reason?